GitWorkflow.md

```diff
@@ -1,9 +1,23 @@
 # Git Workflow
 
+## Branches
+
+Branches are a deviation or split from the main branch that can be adding or removing a specific feature
+For example, I can open a branch to work on a new feature. Since I am on my own branch, I am not interfering
```

Comment:
probably shouldnt split sentences mid line, although limiting line length is nice

Reply:
If we're going to nitpick markdown style you might want to instead standardize on an auto-formatter which we could hook into the project as an action or as a pre-commit hook locally. [Prettier](https://github.com/prettier/prettier) and [mdformat](https://github.com/executablebooks/mdformat).

Frankly we should probably do the same for our Java (both format and lint).

Reply:
For markdown ive been using the markdownlint extension on vscode with default settings, although it doesn't enforce line lengths.

For Java our repo is set up to use spotless as per [this](https://docs.wpilib.org/en/stable/docs/software/advanced-gradlerio/code-formatting.html) wpilib tutorial